src/libsync/pushnotifications.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PUSHNOTIFICATIONS_H
+#define PUSHNOTIFICATIONS_H
+
+#include "websocket.h"
+
+namespace OCC {
+
+class Account;
+class AbstractCredentials;
+
+class PushNotifications : public QObject
+{
+    Q_OBJECT
+
+public:
+    // Question: Is it okay to pass a raw account pointer? Because PushNotifications don't really own the account. The PushNotifications
```

Comment:
It's non-owning so this is likely fine. You *might* want to store it into a QPointer locally if you got concerns about the destruction order and then check for its validity before each use.